File: Julia/pi_sequential.jl
#! /usr/bin/env julia

#  Calculation of π using quadrature. Sequential algorithm using a for loop.
#
#  Copyright © 2012–2013  Russel Winder

require("output.jl")

function execute()
    n = 1000000000
    delta = 1.0 / n
    startTime = time()
    sum = 0.0
    for i = 1:(n + 1)
        sum += 1.0 / (1.0 + ((i - 0.5) * delta) ^ 2)
    end
    pi = 4.0 * delta * sum
    elapseTime = time() - startTime
    out("pi_julia_sequential", pi, n, elapseTime)
end

execute()

# No observable warm-up effect for this code.

File: Nim/pi_sequential.nim
#  Calculation of π using quadrature. Sequential algorithm using a for loop with a range.
#
#  Copyright © 2015 Russel Winder

from output import out_s
from times import epochTime

when isMainModule:
  const n = 100000000 # 10 times fewer due to speed problems
  const delta = 1.0 / n
  let startTime = epochTime()
  var sum:float = 0.0
  for i in 1 .. n:
    let x = (toFloat(i) - 0.5) * delta
    sum += 1.0 / (1.0 + x * x)
  let pi = 4.0 * delta * sum
  let elapseTime = epochTime() - startTime
  out_s("Sequential", pi, n, elapseTime)
